libc/src/unistd/linux/fpathconf.cpp

```diff
@@ -0,0 +1,60 @@
+//===-- Linux implementation of fpathconf ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/unistd/fsync.h"
+
+#include "src/__support/OSUtil/syscall.h" // For internal syscall function.
+#include "src/__support/common.h"
+
+#include "src/errno/libc_errno.h"
+#include <sys/syscall.h> // For syscall numbers.
+
+namespace LIBC_NAMESPACE {
+
+static long pathconfig(const struct statvfs &s, int name) {
+  switch (name) {
+  case _PC_LINK_MAX:
+    return _POSIX_LINK_MAX;
+
+  case _PC_MAX_CANON:
+    return _POSIX_MAX_CANON;
+
+  case _PC_MAX_INPUT:
+    return _POSIX_MAX_INPUT;
+
+  case _PC_NAME_MAX:
+    return _POSIX_NAME_MAX;
+
+  case _PC_PATH_MAX:
+    return _POSIX_PATH_MAX;
+
+  case _PC_PIPE_BUF:
+    return _POSIX_PIPE_BUF;
+
+  case _PC_CHOWN_RESTRICTED:
+    return _POSIX_CHOWN_RESTRICTED;
+
+  case _PC_NO_TRUNC:
+    return _POSIX_NO_TRUNC;
+
+  case _PC_VDISABLE:
+    return _POSIX_VDISABLE;
```

Comment:
We don't define any of these `_POSIX_` prefixed constants anywhere, so for fullbuild mode this will fail to compile.

Reply:
what would be a good alternative to those constants?

Reply:
For `_PC_NAME_MAX` for example. `f_namelen` can be used. For `_POSIX_LINK_MAX`, the maximum allowed links should be decided per `f_type`.

Reply:
@SchrodingerZhu I am having a hard time tryng to match the constants to these fields of of statfs, for example _PC_MAX_CANON, and other fields too. Can you provide me some pointers on this? Thanks. 

https://man7.org/linux/man-pages/man2/statfs.2.html

Reply:
AFAIK, not all fields are directly provided by `statfs`. You can just do the best estimation you have. For example, for name length, `statfs` give you the exact answer, so you can go ahead and use it. For other situation, you may just fallback to POSIX constants. https://android.googlesource.com/platform/bionic/+/refs/heads/main/libc/bionic/pathconf.cpp.

Some tedious parts are that you may need to take a look of the configuration per file system.